Framework/Algorithms/src/AsymmetryCalc.cpp

```diff
@@ -125,14 +125,12 @@ void AsymmetryCalc::exec() {
 
   const size_t blocksize = inputWS->blocksize();
   assert(tmpWS->blocksize() == blocksize);
-  const bool isInputHistogram = inputWS->isHistogramData();
 
   // Create a point data workspace with only one spectra for forward
   API::MatrixWorkspace_sptr outputWS = API::WorkspaceFactory::Instance().create(
       inputWS, 1, blocksize, blocksize);
 
-  // Get the reference to the input x values.
-  auto &tmpX = tmpWS->readX(forward);
+  outputWS->setX(0, tmpWS->histogram(forward).points().cowData());
```

Comment:
This messy syntax is temporary and will go away once I introduced `Histogram::setPoints(const Points &points);`